service/src/main/java/org/cbioportal/genome_nexus/service/internal/SelectedAnnotationServiceImpl.java

```diff
@@ -0,0 +1,145 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package org.cbioportal.genome_nexus.service;
+
+import org.cbioportal.genome_nexus.component.annotation.NotationConverter;
+import org.cbioportal.genome_nexus.model.GenomicLocation;
+import org.cbioportal.genome_nexus.model.VariantAnnotation;
+import org.cbioportal.genome_nexus.service.GenomicLocationAnnotationService;
+import org.cbioportal.genome_nexus.service.exception.VariantAnnotationNotFoundException;
+import org.cbioportal.genome_nexus.service.exception.VariantAnnotationQueryMixedFormatException;
+import org.cbioportal.genome_nexus.service.exception.VariantAnnotationWebServiceException;
+import org.cbioportal.genome_nexus.service.internal.HgvsVariantAnnotationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+import org.apache.commons.logging.LogFactory;
+import org.apache.commons.logging.Log;
+
+@Service
+public class SelectedAnnotationServiceImpl implements SelectedAnnotationService {
+    private final VariantAnnotationService verifiedHgvsVariantAnnotationService;
+    private final GenomicLocationAnnotationService genomicLocationAnnotationService;
+    private final NotationConverter notationConverter; 
+    private final Boolean isRegionAnnotationEnabled; 
+
+private static final Log LOG = LogFactory.getLog(SelectedAnnotationServiceImpl.class);
+
+    @Autowired
+    public SelectedAnnotationServiceImpl(
+            VariantAnnotationService verifiedHgvsVariantAnnotationService,
+            GenomicLocationAnnotationService genomicLocationAnnotationServiceImpl,
+            NotationConverter notationConverter,
+            @Value("${gn_vep.region.url:}")
+            String vepRegionUrl) {
+        this.verifiedHgvsVariantAnnotationService = verifiedHgvsVariantAnnotationService;
+        this.genomicLocationAnnotationService = genomicLocationAnnotationServiceImpl;
+        this.notationConverter = notationConverter;
+        this.isRegionAnnotationEnabled = vepRegionUrl != null && vepRegionUrl.length() > 0;
+    }
+
+    @Override
+    public VariantAnnotation getAnnotation(
+            String variant,
+            String isoformOverrideSource,
+            Map<String, String> token,
+            List<String> fields) throws VariantAnnotationNotFoundException, VariantAnnotationWebServiceException {
+        if (needToConvertHgvsToRegionForAnnotation(variant)) {
+            GenomicLocation variantAsGenomicLocation = notationConverter.hgvsgToGenomicLocation(variant);  
+            // TODO: we should provide a getAnnotation() call to genomicLocationAnnotationService which accepts GenomicLocation
```

Comment:
perhaps the other Genome Nexus developers can help answer these two questions?

Reply:
that would be great. A little more detail for line 82 : the genomicLocationAnnotationService accepts *string* representations of GenomicLocation objects, but maybe it would be more direct / unified if we did not need to convert to a string here before sending the request. (which will get converted back into GenomicLocation in the lower level service)
But the bigger question is whether we wish to support non-HGVS queries coming in to the /annotation endpoint. It seems like we used to, and kind of still do .. but it is not clear whether it still works. It seems to be that when the backend is genome-nexus-vep we would need to convert these other supported formats into region format if we want to submit them to the VEP command line tool.